random.cpp

```diff
@@ -30,11 +30,17 @@ Random::~Random()
 /**
  * Take a reading from the random "sensor"
  */
-Reading	Random::takeReading()
+std::vector<Reading*>* Random::takeReading()
```

Comment:
Function name should also be changed to indicate multiple readings being generated and returned. May be "takeReadings"?